common/src/main/scala/uk/ac/wellcome/platform/stacks/common/services/CatalogueService.scala

```diff
@@ -74,16 +71,23 @@ class CatalogueService(
       // Ensure we are only matching items that match the passed id!
       filteredItems = identifier match {
         case CatalogueItemIdentifier(id) =>
-          items.filter(_.id.catalogueId.value == id)
+          items.filter(_.catalogueId.value == id)
         case SierraItemIdentifier(id) =>
-          items.filter(_.id.sierraId.value == id)
+          items.filter(_.sierraId.value == id)
         case StacksWorkIdentifier(id) =>
-          items.filter(_.id.catalogueId.value == id)
+          items.filter(_.catalogueId.value == id)
         case _ => items
       }
 
-    } yield filteredItems match {
-      case List(item) => Some(item)
-      case _          => None
-    }
+      distinctFilteredItems = filteredItems.distinct
+
+    } yield
+      distinctFilteredItems match {
+        case List(item) => Some(item)
+        case Nil        => None
+        case _ =>
+          throw new Exception(
+            s"Failed to find distinct items in: $distinctFilteredItems"
```

Comment:
Let's be a bit more explicit here:

```suggestion
            s"Found multiple matching items for $identifier in: $distinctFilteredItems"
```